vignettes/pjnz.Rmd

```diff
@@ -0,0 +1,363 @@
+---
+title: "PJNZ Files"
+author: "Robert Ashton"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{PJNZ Files}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+
+[Spectrum](https://www.avenirhealth.org/software-spectrum.php) stores projection data as `PJNZ` files. These are just zips containing several files of projection data.
+
+```{r}
+unzip(system.file("testdata", "Botswana2018.PJNZ", package = "specio"), 
+      list = TRUE)
+```
+
+## Files accessed by `specio`
+
+Note that large portions of the general information about the files are taken from a write up of the Spectrum-EPP Communications between Jeff Eaton and Tim Brown in December 2016. Where this relates to `specio` this information has been updated to reflect the best of current knowledge.
+
+### `.DP`
+The `DP` file contains demographic projection data. The data is persisted as a `csv` with a column containing tags used to identify the field.
+
+The `csv` contains 4 named columns, `Tag`, `Description`, `Notes` and `Data`. There are also an arbitrary number of extra columns all of which contain more data.
+
+```{r}
+lines <- specio:::get_dp_data(
```

Comment:
the use of `lines` here is a bit jarring in the output because these are not really lines. Later on the blocks don't have code above them (I see they're hard coded though) - you could emulate this with `echo = FALSE`

Reply:
Yeah agreed there, I've fixed variable name and also hidden the code. Seeing those particular lines of code doesn't help the vignette at all - they are only wanted for the output.